stakingModel/test/BasicCalculation.test.js

```diff
@@ -12,144 +12,131 @@ contract("InterestDistribution - Basic calculations", ([user1]) => {
-      * AvgYieldRatePerToken = [(AvgYieldRatePerToken(P) x TotalStaked) + (AccumulatedYieldPerToken(P) x Staking x (1-%Donation))]/TotalStaked
+      * GlobalYieldPerToken = GlobalYieldPerToken(P) + GDEarnedInterest/GlobalTotalEffectiveStake.
+      * GDYieldRate = [GDYieldRate(P) + (AccumulatedYieldPerToken(P) x EffectiveStake)]
       * consider :
-      * IntrestTokenRate = 1.02
-      * LastTokenRate = 1
-      * IntrestTokenHoldings = 30 x 1e18
-      * GlobalTotalStake = 20 x 1e18
-      * AccumulatedYieldPerToken(P) = 3
-      * AvgYieldRatePerToken(P) = 0
+      * GlobalTotalEffectiveStake = 20 x 1e18
+      * AccumulatedYieldPerToken(P) = 0
+      * GDYieldRate(P) = 0
       * Staking = 10 x 1e18
       * Donation% = 30%
-      * TotalStaked = 20 x 1e18
-      * output = 3
+      * GDEarnedInterest = 10000 x 1e2
+      * GDEarnedInterestEarned = 10000 x 1e2
       */
 
       await interestDistribution
-        .stakeCalculation(user1, web3.utils.toWei("10", "ether"), 30, web3.utils.toWei("1.02", "ether"), web3.utils.toWei("30", "ether"))
+        .stake(user1, web3.utils.toWei("10", "ether"), 30)
         .catch(e => e);
 
       let yieldData = await interestDistribution.getYieldData(user1);
 
       /**
       * Formula:
-      * GlobalYieldPerToken = GlobalYieldPerToken(P) + [(IntrestTokenRate - LastTokenRate) x IntrestTokenHoldings]/GlobalTotalStake
-      * AvgYieldRatePerToken = [(AvgYieldRatePerToken(P) x TotalStaked) + (AccumulatedYieldPerToken(P) x Staking x (1-%Donation))]/TotalStaked
-      * 0 + [(1.02 - 1) x 30]/20 = [0.02 x 30]x100/20 = 3
-      * (3*10*1e18*(100-30)%)/(20*1e18) = (30*70%)/20 =  1.05 = 1
+      * GlobalYieldPerToken = GlobalYieldPerToken(P) + GDEarnedInterest/GlobalTotalEffectiveStake.
+      * GDYieldRate = [GDYieldRate(P) + (AccumulatedYieldPerToken(P) x EffectiveStake)]
+      * 0 + (10000)/20 = 500 => 500 x 1e11 (27 + 2(G$ precision) - 18(token decimal) = 11 precision points)
+      * 0 + (500 * 7) = 3500 => 3500 x 1e29 (27 + 2(G$ precision) = 29 precision points)
       */      
-      expect(yieldData[0].toString()).to.be.equal("3");
-      expect(yieldData[1].toString()).to.be.equal("1");
+      expect((yieldData[0]/1e11).toString()).to.be.equal("500");
+      expect((Math.round(yieldData[1]/1e29)).toString()).to.be.equal("3500");
   });
 
-  it("Set Accumulated Yield Per Token and Avg Yield Rate Per Token", async () => {
+  it("Set Accumulated Yield Per Token", async () => {
 
 
     /**
-      * Stake = 5 x 10^18
-      * Donation = 0%
-      * IntrestTokenRate = 1.03
-      * LastTokenRate = 1.02
-      * IntrestTokenHoldings = 60 x 1e18
+      * GDEarnedInterest = 270 x 1e2
+      * GDEarnedInterestEarned = 270 x 1e2
       */
     await interestDistribution
-        .stakeCalculation(user1, web3.utils.toWei("6", "ether"), 0, web3.utils.toWei("1.03", "ether"), web3.utils.toWei("60", "ether"))
+        .updateGlobalGDYieldPerToken(270 * 100, 270 * 100)
```

Comment:
not logical that they are same number

Reply:
Isn't it possible if all the stakers are donating 0%?

Reply:
check what are the inputs to updateGlobalGDYieldPerToken, your question is irrelevant

Reply:
Yeah, i know the inputs to updateGlobalGDYieldPerToken() are "Interest earned in G$ after donation" & "Interest earned in G$ before donation". Total donated is 0 till now so I don't understand why both values will not be same.Please elaborate.

Reply:
* @param _blockGDInterest           Interest earned in G$ in  exchange for _blockInterestTokenEarned (after donations)
     * @param _blockInterestTokenEarned  Interest token earned (before donations)

one is in GD the other is in InterestToken

Reply:
interest is earned in interetToken, which is sent to reserve, GD is returned after deducting donations that stay in the reserve

Reply:
ohh, i thought that both are in GoodDollar just that one is total interest generated and other is interest after removing donation amount. Will fix that and let you know. @sirpy

Reply:
Just clarify one thing..... now _blockGDInterest says (after donations), now if 100 GD is newly minted for interest and from that 100GD, 40 is suppose to go to ubi. now what will be its value? 100 or 60?